### PR TITLE
Create paper list post method

### LINKED ISSUE #142: POST method to Create Paper List
### Issue Description

An api POST method will be implemented to create a paper list with a name.

### Steps

- [x] Implementation of method
- [x] Implementation of tests
- [ ] Tests
- [ ] Review

### Deadline for the Issue

12.5.2023

### PR DESCRIPTION
**Pull request by uyduranomer**
In this pull request the POST method to create a new paper list was implemented.
The merge of this pr closes #142

### CODE REVIEW

File: practice_app/api/views.py [hunk 1]
```diff
@@ -455,6 +455,38 @@ def user_registration(request):
...
+        user = authenticate(request, username=username, password=password)
+        if user == None:
+            # If the authentication is failed return an error
+            return JsonResponse({'status' : 'Incorrect username or password!'}, status=401)
+    
+    # Get the paper name with the POST method
+    try:
+        list_title = request.headers['title']
+    except KeyError:
+        return JsonResponse({'status': 'Paper name must be provided!'}, status=400)
```
Comment (hakanaktas0) [inline practice_app/api/views.py:481]: I think is PaperList title, not paper name.

**Review by Bera0422 — APPROVED** [review]
Nice work. The implementation seems good to be merged.

File: practice_app/api/views.py [hunk 2]
```diff
@@ -455,6 +455,38 @@ def user_registration(request):
...
+        password = request.headers['password']
+
+        user = authenticate(request, username=username, password=password)
+        if user == None:
+            # If the authentication is failed return an error
+            return JsonResponse({'status' : 'Incorrect username or password!'}, status=401)
+    
+    # Get the paper name with the POST method
+    try:
+        list_title = request.headers['title']
```
Comment (hakanaktas0) [inline practice_app/api/views.py:479]: you should take this from body, not header.

File: practice_app/api/tests.py [hunk 3]
```diff
@@ -423,6 +424,42 @@ def tearDown(self):
...
+        self.c = Client()
+        self.user = User.objects.create_user(
+            username='testuser', password='testpass'
+        )
+
+    def tearDown(self): # When the tests are completed
+        print("POST method tests for creating paper list completed!")
+
+    def test_for_success(self): # Test for successful case
+        h = {"username": "testuser", "password": "testpass", "title": "testlistname1"}
```
Comment (hakanaktas0) [inline practice_app/api/tests.py:439]: you should give title in body, not header.

**Review by hakanaktas0 — CHANGES_REQUESTED** [review]

File: practice_app/api/views.py (hunk 1 above)
Comment (uyduranomer) [reply practice_app/api/views.py]: fixed

File: practice_app/api/tests.py (hunk 3 above)
Comment (uyduranomer) [reply practice_app/api/tests.py]: fixed

**Review by hakanaktas0 — APPROVED** [review]
Looks good to merge.

**Pull request merged**